ddtrace/_hooks.py

```diff
@@ -20,7 +22,7 @@ def on_request(span, request, response):
             pass
     """
 
-    _hooks = attr.ib(init=False, factory=lambda: collections.defaultdict(set))
+    _hooks = attr.ib(init=False, factory=lambda: collections.defaultdict(set))  # type: Dict[str, Set]
```

Comment:
I think you can do this instead:
```suggestion
    _hooks = attr.ib(init=False, factory=lambda: collections.defaultdict(set), type=Dict[str, Set])
```
And it'd actually be wrong as `Dict` and `defaultdict` are different. There's a type for `defaultdict`:

https://docs.python.org/3/library/typing.html#typing.DefaultDict

Reply:
Thanks for the catch! 😄